snap/snapcraft.yaml

```diff
@@ -1,160 +1,72 @@
-name: openvino-toolkit-2204
-base: core22
+name: openvino-toolkit-2404
+base: core24
+version: '2024.4.0'
 summary: Intel's OpenVINO Toolkit
 description: |
-  SDK to build OpenVINO applications
+  OpenVINO™ is an open-source toolkit for optimizing and deploying AI inference.
 
 grade: devel
 confinement: strict
 adopt-info: openvino
 
-architectures:
-  - build-on: amd64
+platforms:
+  amd64:
 
-# the recommended mountpoint for that content is /openvino
+# the recommended mountpoint for the content is /openvino
 slots:
-  openvino-toolkit-2204:
+  openvino-libs:
     interface: content
+    content: openvino-libs-2404
     read:
       - /
 
 parts:
-  cmake:
-    source: https://github.com/Kitware/CMake/releases/download/v3.17.2/cmake-3.17.2.tar.gz
-    plugin: nil
-    override-build: |
-      ./bootstrap --prefix=/usr
-      make -j4
-      make install DESTDIR=$SNAPCRAFT_PART_INSTALL
-    build-environment:
-      - PATH: /usr/lib/ccache:$PATH
-    build-packages:
-      - ccache
-      - build-essential
-      - libssl-dev
   openvino:
+    source-type: git
     source: https://github.com/openvinotoolkit/openvino.git
-    source-branch: "releases/2024/0"
+    source-tag: 2024.4.0
     plugin: cmake
-    build-environment:
-      - OpenCV_DIR: $SNAPCRAFT_STAGE/opencv
-      - PATH: /usr/lib/ccache:$PATH
+    cmake-parameters:
+      - -DENABLE_PYTHON=ON
+      - -DPython3_EXECUTABLE=/usr/bin/python3.12
+      - -DCMAKE_INSTALL_PREFIX=/usr
     override-pull: |
       craftctl default
-      craftctl set version="$(git describe --tags --abbrev=4 --always)"
```

Comment:
I would prefer setting the snap version based on the tag + hash to help distinguish between multiple snap revisions built from the same upstream openvino tag.

Reply:
Done!